src/Microsoft.SqlTools.Dmp.Contracts/Hosting/InitializeRequest.cs

```diff
@@ -0,0 +1,61 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace Microsoft.SqlTools.Dmp.Contracts.Hosting
+{
+    public class InitializeParams
+    {
+        /// <summary>
+        /// Capabilities of the client that is initializing this service host
+        /// </summary>
+        public ClientCapabilities Capabilities { get; set; }
+        
+        /// <summary>
+        /// ID of the process that is initializing this service host
+        /// </summary>
+        public int ProcessId { get; set; }
+        
+        /// <summary>
+        /// Root path of the editor's open workspace. If null, it is assumed that a file was opened
+        /// without having a workspace open.
+        /// </summary>
+        /// <remarks>If both RootPath and RootUri are available, RootUri is preferred</remarks>
+        [Obsolete]
```

Comment:
Good to mention why this is now obsolete... where is it used then?

Reply:
As per VSCode's language service protocol, RootPath is deprecated in favor of RootUri, which is kinda sorta deprecated in favor of the not-completely-finalized workspace folders. Right now the workspace service uses these, iirc.